seaborn/matrix.py

```diff
@@ -168,6 +168,19 @@ def __init__(self, data, vmin, vmax, cmap, center, robust, annot, fmt,
         self.data = data
         self.plot_data = plot_data
         self.annot = annot
+
+        if isinstance(self.annot, bool) and self.annot:
+            self.annot_data = plot_data
+        elif not isinstance(self.annot, bool):
```

Comment:
Oh I see, the default is `False` not `None`. What do you think about changing to `None`? It will make this logic a little messier, but it might be a stronger signal that values other than `True` or `False ` are valid for this parameter.